ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -269,12 +273,14 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
       for (int i = 0; i < readSet.count(); i++) {
         messageInfoList.add(indexMessages.get(readSet.getKeyAt(i)));
       }
+      onSuccess();
       return new StoreInfo(readSet, messageInfoList);
-    } catch (StoreException e) {
-      throw e;
     } catch (Exception e) {
-      throw new StoreException("Unknown exception while trying to fetch blobs from store " + dataDir, e,
-          StoreErrorCodes.Unknown_Error);
+      if (e instanceof StoreException && ((StoreException) e).getErrorCode() == StoreErrorCodes.IOError) {
+        onError();
+      }
+      throw new StoreException("Exception while trying to fetch blobs from store " + dataDir, e,
```

Comment:
this might rewrap a `StoreException`. Isn't it better to change the existing code in the following way?
```
} catch (StoreException e) {
   if (e.getErrorCode().equals(StoreErrorCodes.IOError) {
     onError();
   }
   throw e;
} catch Exception {
  ...
}
```
Also, don't you also need to check for the `IOException` and the string expected? (I think there should be a function for that)

Reply:
Right, changed this piece of code to catch different types of exception separately. By the way, this function doesn't throw `IOException`. So, there is no need to capture it.